File: GasInSight.App/src/types/sensor.ts
export enum SensorType {
  Temperature = "Temperature",
  Humidity = "Humidity",
  Pressure = "Pressure",
  Flow = "Flow",
  Volume = "Volume",
  GasComposition = "GasComposition",
  LiquidComposition = "LiquidComposition",
  Vibration = "Vibration",
  Noise = "Noise",
  Corrosion = "Corrosion",
  GasDetection = "GasDetection",
  FlameDetection = "FlameDetection",
  LevelIndicator = "LevelIndicator",
  ValveStatus = "ValveStatus",
  PumpStatus = "PumpStatus",
  CompressorStatus = "CompressorStatus",
  PowerConsumption = "PowerConsumption",
  WaterContent = "WaterContent",
  OxygenContent = "OxygenContent",
  HydrogenSulfideContent = "HydrogenSulfideContent",
  CarbonDioxideContent = "CarbonDioxideContent",
  ParticulateMatter = "ParticulateMatter",
  LeakDetection = "LeakDetection"
}

export enum SensorStatus {
  Active = "Active",
  Inactive = "Inactive",
  Maintenance = "Maintenance",
  Fault = "Fault"
}

export interface Sensor {
  id: string;
  name: string;
  facilityId: string;
  locationId?: number;
  status: SensorStatus;
  installedAt: string;
  type: SensorType;
  location?: {
    id: number;
    address?: string;
    longitude: number;
    latitude: number;
  };
}

export interface SensorRecord {
  sensorId: string;
  trackedAt: string;
  data: string;
}

File: GasInSight.Server/app/models/sensor.py
from sqlalchemy import Column, String, Integer, ForeignKey, DateTime, Enum
from sqlalchemy.orm import relationship
import enum
from app.database import Base


class SensorType(enum.Enum):
    Temperature = "Temperature"
    Humidity = "Humidity"
    Pressure = "Pressure"
    Flow = "Flow"
    Volume = "Volume"
    GasComposition = "GasComposition"
    LiquidComposition = "LiquidComposition"
    Vibration = "Vibration"
    Noise = "Noise"
    Corrosion = "Corrosion"
    GasDetection = "GasDetection"
    FlameDetection = "FlameDetection"
    LevelIndicator = "LevelIndicator"
    ValveStatus = "ValveStatus"
    PumpStatus = "PumpStatus"
    CompressorStatus = "CompressorStatus"
    PowerConsumption = "PowerConsumption"
    WaterContent = "WaterContent"
    OxygenContent = "OxygenContent"
    HydrogenSulfideContent = "HydrogenSulfideContent"
    CarbonDioxideContent = "CarbonDioxideContent"
    ParticulateMatter = "ParticulateMatter"
    LeakDetection = "LeakDetection"


class SensorStatus(enum.Enum):
    Enabled = "Enabled"
    Disabled = "Disabled"
    Removed = "Removed"


class Sensor(Base):
    __tablename__ = "sensors"

    id = Column(String, primary_key=True, index=True)
    name = Column(String)
    facility_id = Column(String, ForeignKey("facilities.id"))
    location_id = Column(Integer, ForeignKey("locations.id"))
    status = Column(Enum(SensorStatus))
    installed_at = Column(DateTime)
    type = Column(Enum(SensorType))

    facility = relationship("Facility", back_populates="sensors")
    location = relationship("Location", back_populates="sensors")
    records = relationship("SensorRecord", back_populates="sensor")
